api/src/server.ts

```diff
@@ -17,14 +18,15 @@ app.get('/api', async (req: Request, res: Response) => {
     return;
   }
 
-  const response = await fetch(url, {
-    headers: {
-      'User-Agent':
-        'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3',
-    },
-    redirect: 'follow',
+  const browser = await puppeteer.launch({
+    headless: true,
+    args: ['--no-sandbox', '--disable-setuid-sandbox'],
   });
-  const html = await response.text();
+  const page = await browser.newPage();
+  await page.goto(url, { waitUntil: 'networkidle2' });
+  const html: string = await page.evaluate(() => document.body.innerHTML);
```

Comment:
[memo]
This ended up running javascript in the browser, so there was no change from node-fetch.